### PR TITLE
Bump default julia to 1.10

### PR DESCRIPTION
**Pull request by lgoettgens**
The values in `src/defaults.jl` assume that the corresponding PRs 
- https://github.com/Nemocas/AbstractAlgebra.jl/pull/1543
- https://github.com/Nemocas/Nemo.jl/pull/1609
- https://github.com/thofma/Hecke.jl/pull/1339
- https://github.com/oscar-system/Oscar.jl/pull/3144
- https://github.com/oscar-system/Singular.jl/pull/744

get merged before the next minor release. If not, this PR has to be adapted accordingly.

### CODE REVIEW

File: src/defaults.jl [hunk 1]
```diff
@@ -11,16 +11,21 @@ const default_branches = [ "<matching>", "release" ]
...
-                         v"0.33.8"     => (v"1.8", v"1.9" )],
-    :Singular        => [v"0.18.8"     => (v"1.9", v"1.10" ),
-                         v"0.18.3"     => (v"1.8", v"1.9" )],
+    :Oscar           => [v"0.14.0-DEV" => (v"1.10", v"1.13"),
+                         v"0.13.0-DEV" => (v"1.9" , v"1.11"),
+                         v"0.12.1-DEV" => (v"1.8" , v"1.11")],
+    :Hecke           => [v"0.23.4"     => (v"1.10", v"1.11"),
+                         v"0.19.5"     => (v"1.9" , v"1.10"),
+                         v"0.16.7"     => (v"1.8" , v"1.9" )],
+    :AbstractAlgebra => [v"0.34.8"     => (v"1.10", v"1.11"),
```
Comment (lgoettgens) [inline src/defaults.jl]:
```suggestion
    :AbstractAlgebra => [v"0.35.3"     => (v"1.10", v"1.11"),
```

File: src/defaults.jl [hunk 2]
```diff
@@ -11,16 +11,21 @@ const default_branches = [ "<matching>", "release" ]
...
+                         v"0.31.0"     => (v"1.9" , v"1.10"),
+                         v"0.29.5"     => (v"1.8" , v"1.9" )],
+    :Nemo            => [v"0.38.4"     => (v"1.10", v"1.11"),
```
Comment (lgoettgens) [inline src/defaults.jl]:
```suggestion
    :Nemo            => [v"0.40.0"     => (v"1.10", v"1.11"),
```

File: src/defaults.jl [hunk 3]
```diff
@@ -11,16 +11,21 @@ const default_branches = [ "<matching>", "release" ]
...
-    :AbstractAlgebra => [v"0.31.0"     => (v"1.9", v"1.10"),
-                         v"0.29.5"     => (v"1.8", v"1.9" )],
-    :Nemo            => [v"0.35.1"     => (v"1.9", v"1.10" ),
-                         v"0.33.8"     => (v"1.8", v"1.9" )],
-    :Singular        => [v"0.18.8"     => (v"1.9", v"1.10" ),
-                         v"0.18.3"     => (v"1.8", v"1.9" )],
+    :Oscar           => [v"0.14.0-DEV" => (v"1.10", v"1.13"),
+                         v"0.13.0-DEV" => (v"1.9" , v"1.11"),
+                         v"0.12.1-DEV" => (v"1.8" , v"1.11")],
+    :Hecke           => [v"0.23.4"     => (v"1.10", v"1.11"),
```
Comment (lgoettgens) [inline src/defaults.jl]:
```suggestion
    :Hecke           => [v"0.24.1"     => (v"1.10", v"1.11"),
```

File: src/defaults.jl [hunk 4]
```diff
@@ -11,16 +11,21 @@ const default_branches = [ "<matching>", "release" ]
...
+                         v"0.19.5"     => (v"1.9" , v"1.10"),
+                         v"0.16.7"     => (v"1.8" , v"1.9" )],
+    :AbstractAlgebra => [v"0.34.8"     => (v"1.10", v"1.11"),
+                         v"0.31.0"     => (v"1.9" , v"1.10"),
+                         v"0.29.5"     => (v"1.8" , v"1.9" )],
+    :Nemo            => [v"0.38.4"     => (v"1.10", v"1.11"),
+                         v"0.35.1"     => (v"1.9" , v"1.10"),
+                         v"0.33.8"     => (v"1.8" , v"1.9" )],
+    :Singular        => [v"0.21.2"     => (v"1.10", v"1.11"),
```
Comment (lgoettgens) [inline src/defaults.jl]:
```suggestion
    :Singular        => [v"0.21.3"     => (v"1.10", v"1.11"),
```

**Review by lgoettgens — COMMENTED** [review]
updating all doctest version bounds now that the PRs are merged

File: src/defaults.jl [hunk 5]
```diff
@@ -11,16 +11,21 @@ const default_branches = [ "<matching>", "release" ]
...
-                         v"0.12.1-DEV" => (v"1.8", v"1.11")],
-    :Hecke           => [v"0.19.5"     => (v"1.9", v"1.10" ),
-                         v"0.16.7"     => (v"1.8", v"1.9" )],
-    :AbstractAlgebra => [v"0.31.0"     => (v"1.9", v"1.10"),
-                         v"0.29.5"     => (v"1.8", v"1.9" )],
-    :Nemo            => [v"0.35.1"     => (v"1.9", v"1.10" ),
-                         v"0.33.8"     => (v"1.8", v"1.9" )],
-    :Singular        => [v"0.18.8"     => (v"1.9", v"1.10" ),
-                         v"0.18.3"     => (v"1.8", v"1.9" )],
+    :Oscar           => [v"0.14.0-DEV" => (v"1.10", v"1.13"),
```
Comment (benlorenz) [inline src/defaults.jl]:
```suggestion
    :Oscar           => [v"0.14.0-DEV" => (v"1.10", v"1.12"),
```
We don't know yet if julia 1.12 might be breaking our doctests, so better keep that as < 1.12.

**Review by benlorenz — APPROVED** [review]
thanks

**Pull request merged**